### DISCUSSION TITLE
Rename PropertyIdentifier to be consistent with ASHRAE Standard

### DISCUSSION THREAD

**Discussion by guangyuzhang** [JoelBender/bacpypes#378] [General]
Hello Joel,

As an entry level BACnet developer, I found it's a little hard to read the code against with the Standard because the PropertyIdentifiers are different in them. For example, the code uses 'presentValue', 'statusFlags', the standard uses 'Present_Value', 'Status_Flags'. 

![Screenshot from 2020-12-18 18-15-44](https://user-images.githubusercontent.com/4197169/102602887-15e49d00-415d-11eb-9260-27c5d0ff7cac.png)

I understand it's hard to refactor the code, but it will be helpful.

Thanks,
Guangyu Zhang

**Comment by JoelBender** [comment]
It's worse than that, the Examples uses object types in all uppercase like ANALOG_INPUT and property names in quotes like `Object_Identifier` and Annex Y uses the ASN.1 names like `analog-input` and `present-value`.  There is a basic ambiguity if Object_Identifier would refer to the property name or the class of all object identifiers, and having the "BACnet" prefix in front of all of the class names (itself a convention that was born out of adopting ASN.1 and the subsequent syntax requirements) seemed excessive in the world of object oriented programming.  Of course "-" is not a valid character identifier in a vast majority of the programming languages, and many shops have been opposed to the underscore, which is not as big a deal these days as it used to be (like [encoding the data type as a prefix](https://en.wikipedia.org/wiki/Hungarian_notation) for property/attribute names).

I proposed making an editorial change to the standard to make them all consistent and programmer friendly; **UpperCamelCase** names for things and **lowerCamelCase** names for properties.  Because every API and tool kits makes its own decisions about naming conventions, this was regarded as ultimately a lot of work for very little gain.

Because Annex Y references ASN.1 names, and BACnet Web Services does as well, I also use `present-value` RDF predicate names expecting that it will be a little easier to get approved.  At this point changing the ASN.1 productions would break other conventions, so at this point it's just one of those things to get used to.
